### ISSUE TITLE
Cleanup for 2.2.1 draft spec

### ISSUE THREAD

**Issue by swinslow** [spdx/spdx-spec#454]
__Updated 4__: virtually all of the issues _should_ be fixed after #460 is merged. The only remaining items are:

- [x] **Clause 4.4**: confirm that indented sub-bullets are displaying correctly at https://spdx.github.io/spdx-spec/v2-draft/conformance/ after PR is merged; if they aren't, fix them
- [x] **Annex J.2**: Replace with the actual changes from v2.1 to v2.2 (what's currently shown are the changes from v2.0 to v2.1)

= = =

I'm in the midst of reviewing the draft 2.2.1 SPDX spec at https://spdx.github.io/spdx-spec/v2-draft/ for typos, and in a couple of places for minor stylistic issues.

__Updated__: completed read-through, additional issues noted below. I'll do my best to prepare a PR to fix some or all of these, but may not have it ready before tech team call tomorrow.

__Updated 2__: in process of preparing PR; I'll be checking these off as I go through, but note that they aren't addressed in the main branch until the PR is actually submitted and merged.

__Updated 3__: added items from @zvr below

- [x] **across many sections**: need to align on capitalization for “SPDX Document” vs. “SPDX document”
  - also update in SPDX ontology and JSON schema files? => yes, tracked in #459
- [x] **Copyright** and **Clause 1**: also add (R) after first instance of “Software Package Data Exchange”
- [x] **Foreword**: ISO links don’t work, not absolute
- [x] **Foreword**: Missing close paren in 3rd para
- [x] **Foreword**: Last paragraph: Rex -- Does it need to say only this? Can we change or supplement the language here to clarify that the SPDX spec development occurs within the SPDX workgroup, rather than in ISO itself?
- [x] **Introduction**: “the SPDX working group” => “the SPDX workgroup”
- [x] **Clause 1**: “set of files or snippets” => remove “set of”, already appears before “software packages”
- [x] **Clause 2**: the urls aren’t rendering as links
- [x] **Clause 3**: the “.x” section numbers will need to be filled in
- [x] **Clause 3**: I assume the terms should be alphabetized?
- [x] **Clause 3**: "file information section" includes a reference to "files (3.x)", but "file" is not defined -- should either define "file" and update section numbers / references, or else remove "(3.x)" from end of this line
  - remove parenthetical ref
- [x] **Clause 3**: after prior edits, definitions are now wrapping onto same line as terms; needs extra hard line break after each term
- [x] **Clause 3**: “annotation section” and “relationship section” should probably have “information” before “section”, to parallel the others -- or is there a reason to omit "information" from these?
  - [x] add "information" to definitions
  - [x] add "information" to titles of ann and rln sections -- DON'T change filenames
- [x] **Clause 3**, _document creation information section_: “SPDX file” => “SPDX document”
- [x] **Clause 4.1**: “the SPDX Working Group” => “the SPDX workgroup”
- [x] **Clause 4.1**: brackets around “[Those earlier editions...]” should be removed if finalized
  - not removing, leave as-is
- [x] **Clause 4.1**: at end, stray link says `see also [1]`; what is this meant to link to?
  - looks like this is meant to refer to the bibliography; kept as [1] but changed to link
  - [x] confirm after submitting PR that the link correctly renders, since the brackets around 1 should still appear
- [x] **Clause 4.3**: Is this meant to include a full list of the different values that can appear for Cardinality fields? I note that e.g. several of the Cardinality fields in the document now use “1..1” which is not listed here.
- [x] **Clause 4.4**: bullet points with list of formats needs to be indented another level
  - Note that these are rendering correctly in GitHub markdown view, but not at https://spdx.github.io/spdx-spec/v2-draft/conformance/
- [x] **Clause 4.4**: XML link is not rendering
- [x] **Clause 4.5**: add link to SPDX Trademark Page
- [x] **Clause 4.6**: fix link to Annex H
- [x] **Clause 5.1**: fix link to Annex C and change "data model" to "object model"
- [x] **Clause 5.2.1**: fix link to Clause 6
- [x] **Clause 5.3**: last bullet point, text cut off – add at the end “to be taken.”
- [x] **Clause 7.8.2**, section 2: link to clause 11 is broken
- [x] **Clause 7.13.1**: text and link to 3.15 after bullet points is wrong, should be 7.15
- [x] **Clause 7.21.3**: at end after RDF examples, link to 6.5 is broken
- [x] **Clause 7.22.1**: text and links to 3.21 in Table 34 are wrong
- [x] **global**: do global check for “SPDX file” vs. “SPDX Document”
  - also update in SPDX ontology and JSON schema files? => yes, tracked in #459
- [x] **global**: do global check for “SPDX data” => “SPDX Document”, “SPDX data creator” => “SPDX Document creator”
  - also update in SPDX ontology and JSON schema files? => yes, tracked in #459
- [x] **Clause 8.6.1**: odd line breaking in Format row of Table 41
- [x] **Clause 9.2.1**: end of Format row of Table 53 needs links for subclause references
- [x] **Clause 9.5.1**: “If the Concluded License is not the same as the License Information in File” => “File” should be “Snippet” since this is in the Snippet section
- [x] **Clause 9.5.1**: end of Format row of Table 56 needs change from “Appendix” to “Annex”, and add link
- [x] **Clause 9.6.1**: definitions of `NONE` and `NOASSERTION` should refer to “SPDX document creator” instead of “SPDX snippet creator”
- [x] **Clause 9.7.2**: “. . . if it does not match the License Information in File” => “File” should be “Snippet” since this is in the Snippet section
- [x] **Clause 9.8.1**: definitions of `NONE` and `NOASSERTION` should refer to “snippet” instead of “file”
- [x] **Clause 9.11.1, 9.11.2**: “SPDX data creator” => “SPDX Document creator”
- [x] **Clause 10**: throughout, this refers to using `LicenseRef-` licenses in packages and files, but they can also be used for snippets; should be clarified (either by adding reference to snippets and/or by removing references to packages and files)
- [x] **Clause 11.1.1**: “whether there are other SPDX elements (package/file/snippet) are connected” => add “that” after close paren
- [x] **Clause 11.1.1**: “the keywords `NONE`” => “the keyword `NONE`”
- [x] **Clause 12**: throughout, this refers to using annotations for packages and files, but they can also be used for snippets; should be clarified (either by adding reference to snippets and/or by removing references to packages and files)
- [x] **Annex B.9.2**: add link for URL
- [x] **Annex D.1**: fix broken links / entries for `license-id` and `license-exception-id` in grammar
  - [x] change to code section with backticks
- [x] **Annex D.3**: in third bullet point, “A SPDX” => “An SPDX”
- [x] **Annex D.4.1**: reference in final sentence should be corrected to D.4.5 and changed to a link
- [x] **Annex E.1**: update link in last sentence to .dev site
- [x] **Annex E.2**: fix broken link / entry for `SPDX-License-Identifier`
  - [x] change to code section with backticks
- [x] **Annex E.4**: in first sentence, “a SPDX” => “an SPDX”
- [x] **Annex E.4**: after examples, and again in last paragraph, fix broken links to Annex D
- [x] **Annex E.4**: in next paragraph after that, fix broken link for requesting new license
- [x] **Annex H**: minor grammar fixes / edits
- [x] **Annex I.1**: first para, fix broken link to Annex E
- [x] **Annex I.1**: second para, fix broken link to Clause 8
- [x] **Annex I.2**: second para, fix broken link to File Information
- [x] **Annex J.3**: I assume we want to leave these as pointing to the old section / appendix numberings, since they refer to old versions of the spec?
- [x] **Bibliography**: in 1, update spdx.org to spdx.dev
- [x] **Bibliography**: reference 2 does not appear to be used anywhere (at least, there isn't a `[2]` elsewhere in the repo); should this be deleted?
- [x] **rationale.md**: delete, no longer used
- [x] B.15.2 - fix backslashes by putting in backticks

from @zvr:
- [x] B.15.1 the address of the git repository should be spelled out, since the link will not be visible
- [x] B.15.2 the whole template format does not match what we currently have and use...
  - discussed on 2020-07-07 call, will update this but not for 2.2.1 since it should match substance of 2.2
  - separately consider whether to deprecate template format / switch to standardizing XML - as 3.0
- [x] D.1 (you have to fix the links on your list) license-exception-id should point to A.2, not B.2

**Comment by goneall** [comment]
@swinslow My bad on the SPDX document capitalization.  See issue #448 

If you're going to do other updates to the text, please go ahead and fix the capitalization too - otherwise we'll cause merge conflicts.

**Comment by swinslow** [comment]
Thanks @goneall -- if I am able to get to handling the other issues above prior to the tech call tmw, I'll try to fix that as well.

**Comment by swinslow** [comment]
@RexJaeschke Question for you -- in the foreword, the last two paragraphs currently say:

> This document was prepared by the SPDX workgroup, which is hosted by The Linux Foundation. (That workgroup included representatives from more than 20 organizations involved with software development, systems and tool vendors, foundations and systems integrators.) This document was adopted, under the PAS procedure, by Joint Technical Committee ISO/IEC JTC 1, Information technology, in parallel with its approval by the national bodies of ISO and IEC.

> Any feedback or questions on this document should be directed to the user’s national standards body. A complete listing of these bodies can be found at [https://www.iso.org/members.html](https://www.iso.org/members.html).

Will ISO require the last paragraph to say exactly this? Or would it be okay to also add something like:

> Questions and proposals for changes can also be directed to the SPDX workgroup, by submitting issues and pull requests to the SPDX specification repository at [https://github.com/spdx/spdx-spec](https://github.com/spdx/spdx-spec).

Ideally I'd like to be able to keep it clear that people can continue to join and participate in the SPDX workgroup without having to contact their national standards body first...

**Comment by RexJaeschke** [comment]
@swinslow As we're producing an ISO standard, we're required to use the boilerplate text from their Word template, as I have done. Any question/error found in an ISO spec then gets reported as a Defect Report (DR) through an ISO (or ISO/IEC JTC 1) National Body (country, that is) will (eventually) get back to the PAS submitter. That said, LF will be co-publishing this standard, and in the LF version you can say whatever you like about how maintenance works, how to contribute, who the contributors were, and so on. 

Bottom line, as LF will continue co-publishing this spec, you'll have two flavors: ISO and LF. So, you can maintain the md in one of those flavors and spin-off the other one at the time of publication. For now, I have the md in ISO flavor, but going forward, you might want to reverse that. Either way, you'll have some work to do each edition to generate both flavors.

**Comment by swinslow** [comment]
Perfect, thanks @RexJaeschke! Makes sense and understood.

**Comment by zvr** [comment]
Shall I add a couple of entries on this issue, instead of opening a new one?

- [ ] **B.15.1** the address of the git repository should be spelled out, since the link will not be visible
- [ ] **B.15.2** the whole template format does not match what we currently have and use...
- [ ] **D.1** (you have to fix the links on your list) `license-exception-id` should point to A.2, not B.2

**Comment by swinslow** [comment]
@zvr added to list above -- thank you!

**Comment by kestewart** [comment]
Annex J.2 is solved by PR# 461,  and manually confirmed indents are in place.    So we're done!
